include/vcpkg/base/files.h

```diff
@@ -246,6 +246,16 @@ namespace vcpkg
         void rename_with_retry(const Path& old_path, const Path& new_path, std::error_code& ec) const;
         void rename_with_retry(const Path& old_path, const Path& new_path, LineInfo li) const;
 
+        // Rename old_path -> new_path, but consider new_path already existing as acceptable.
+        // Traditionally used to interact with downloads, or git tree cache, where multiple
+        // instances of vcpkg may be trying to do the same action at the same time.
+        //
+        // Returns whether the rename actually happened.
+        //
+        // If `old_path` and `new_path` resolve to the same file, the behavior is undefined.
+        bool rename_cas_like(const Path& old_path, const Path& new_path, std::error_code& ec) const;
+        bool rename_cas_like(const Path& old_path, const Path& new_path, LineInfo li) const;
```

Comment:
`cas`? Maybe choose a different name 😅

Reply:
I considered something like `xchg` but that would imply the target is overwritten if it exists. I'm happy to use a different name but I don't have ideas for anything better.

Reply:
`rename_to_or_delete_if_exists` 😄

Reply:
I feel like that name is a bit too close to just 'putting the code entirely in the name' but when I don't feel strongly about things and someone felt strongly enough about it to make a comment, I'm going to apply the suggestion.

Reply:
How about `rename_or_delete`? The 'to' part is already implied by the rename, and avoids 'delete if exists' sounding like 'delete the target if it exists'. This way, one of those two verbs are clearly happening to the same subject (the source); we either rename the thing, or if we can't, we delete it.

Reply:
Sounds good!